src/levanter/models/qwen.py

```diff
@@ -287,6 +365,61 @@ def init(cls, Vocab: Axis, config: QwenConfig, *, key) -> "QwenLMHeadModel":
 
         return QwenLMHeadModel(transformer, embeddings, lm_head)
 
+    def initial_cache(self, spec: PageTableSpec, *, dtype) -> ListCache[KvPageCache]:
```

Comment:
I noticed that initial_cache and decode are directly copied from https://github.com/marin-community/levanter/blob/77aa5913d28590c7b55e6ae5e004b4da5d4ba22d/src/levanter/models/llama.py#L510. Can we maybe inherit from `LlamaLMHeadModel` to reduce code duplication?